src/TestFramework/TestFramework/Attributes/DataSource/TestIdGenerationStrategy.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+/// <summary>
+/// The supported test ID generation strategies.
+/// </summary>
+public enum TestIdGenerationStrategy
+{
+    /// <summary>
+    /// Uses legacy test ID generation. <see cref="ITestDataSource"/> tests will not be discovered and instead they will be collapsed into one parent test.
+    /// </summary>
+    /// <remarks>
+    /// This option is incompatible with <see cref="TestDataSourceDiscoveryOption.DuringDiscovery"/> option of <see cref="TestDataSourceDiscoveryAttribute"/> and will be ignored.
+    /// This was the default option until version 2.2.3.
+    /// </remarks>
+    [Obsolete("This strategy is provided to reduce impact on existing users. It will be removed in a future release.")]
+    Legacy = 0,
```

Comment:
Marked as obsolete so that we can drop it in a near future because we mainly only want to ease migration but not support all these ID generators.

Reply:
Setting this to 0 makes it default value when it's not set. Do we want that?

Reply:
I have run some manual checks and added some integration tests and I confirm that having `Legacy` as `0` is not a problem. When nothing is setup we correctly fallback to default being `FullyQualified`.